### ISSUE TITLE
new_mail_command only sends notification once and never again

### ISSUE THREAD

**Issue by eralwen** [neomutt/neomutt#2003]
## Expected Behaviour
I am supposed to get new notifications every time there's new email with my notification command

## Actual Behaviour
The new notifications only works once and it is always the first time. So it works for the first time and never again after. One thing I think I may have noticed is that I see the new mail in inbox below the status line during the first time, but I never the see it again after. If I restart neomutt, it works again but like before, only once. 

## Steps to Reproduce
Here is my setting for new_mail_command:
`set new_mail_command="dunstify -r 1 'New Email'" `
dunstify is like a notify-send. I also done this with notify-send with the literally the exact same set_mail_command on the website, but the same thing happens: it works one time and not again

## How often does this happen?

- Always

## When did it start to happen?
I am a new user, start two days ago.  I just have a basic config with normal IMAP and SMTP, and some changed keybindings. 

## NeoMutt Version

```
NeoMutt 20191111
Copyright (C) 1996-2016 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 5.3.11-arch1-1 (x86_64)
ncurses: ncurses 6.1.20180127 (compiled with 6.1.20180127)
libidn2: 2.3.0 (compiled with 2.2.0)
GPGme: 1.13.1
hcache backends: kyotocabinet, gdbm, lmdb

Compiler:
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /build/gcc/src/gcc/configure --prefix=/usr --libdir=/usr/lib --libexecdir=/usr/lib --mandir=/usr/share/man --infodir=/usr/share/info --with-bugurl=https://bugs.archlinux.org/ --enable-languages=c,c++,ada,fortran,go,lto,objc,obj-c++,d --enable-shared --enable-threads=posix --with-system-zlib --with-isl --enable-__cxa_atexit --disable-libunwind-exceptions --enable-clocale=gnu --disable-libstdcxx-pch --disable-libssp --enable-gnu-unique-object --enable-linker-build-id --enable-lto --enable-plugin --enable-install-libiberty --with-linker-hash-style=gnu --enable-gnu-indirect-function --enable-multilib --disable-werror --enable-checking=release --enable-default-pie --enable-default-ssp --enable-cet=auto gdc_include_dir=/usr/include/dlang/gdc
Thread model: posix
gcc version 9.2.0 (GCC) 

Configure options: --prefix=/usr --sysconfdir=/etc --libexecdir=/usr/lib --gpgme --enable-lua --notmuch --gss --ssl --sasl --with-ui=ncurses --with-idn2=/usr --idn=0 --idn2=1 --lmdb --kyotocabinet --gdbm

Compilation CFLAGS: -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -std=c99 -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -I/usr/include -DNCURSES_WIDECHAR

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  -autocrypt +bkgdset +color +curs_set +fcntl -flock -fmemopen +futimens 
  +getaddrinfo -gnutls +gpgme +gss +hcache -homespool +idn +inotify 
  -locales_hack +lua +meta -mixmaster +nls +notmuch +openssl +pgp +sasl +smime 
  -sqlite +start_color +sun_attachment +typeahead 
MAILPATH="/var/mail"
PKGDATADIR="/usr/share/neomutt"
SENDMAIL="/usr/sbin/sendmail"
SYSCONFDIR="/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>

```

## Extra Info

* Linux 5.3.11-arch1-1 #1 SMP PREEMPT Tue, 12 Nov 2019 22:19:48 +0000 x86_64 GNU/Linux
* Were you using multiple copies of NeoMutt at once? no
* Were you using 'screen' or 'tmux'? no 
* Is your email local (maildir) or remote (IMAP)? remote

**Comment by gahr** [comment]
Does this explain? https://github.com/neomutt/neomutt/issues/1881#issuecomment-556107128

**Comment by eralwen** [comment]
I think my issue relates to part 1. I am testing this by repeatedly sending mail to INBOX, which I believe is my currently selected mailbox (the one that is currently being indexed?). Like the number of new messages should be increasing, because they're all going to INBOX, so I think I should be getting a new notification, but correct me if I am wrong.

**Comment by gahr** [comment]
It should be fixed by 2cc1e12e36ccc370a9c69b7fb529aab849b2e015 then. Can you please give master a try?

**Comment by eralwen** [comment]
I won't try this as I assume the bug is fixed. I will close this now. I will just wait for a new release where this bug fix is implemented. Thanks!
